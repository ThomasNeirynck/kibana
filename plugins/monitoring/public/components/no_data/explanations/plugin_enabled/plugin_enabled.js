import React from 'react';
import PropTypes from 'prop-types';
import { EuiCode } from '@elastic/eui';

export function ExplainPluginEnabled({ context, property, data }) {
  return (
    <p>
      We checked the {context} settings and found that <EuiCode>{property}</EuiCode>{' '}
      is set to <EuiCode>{data}</EuiCode> set, which disables monitoring.
      Removing the <EuiCode>xpack.monitoring.enabled: false</EuiCode> setting
      from your configuration will put the default into effect and enable Monitoring.
    </p>
  );
}

ExplainPluginEnabled.propTypes = {
  property: PropTypes.string.isRequired,
  context: PropTypes.string.isRequired,
  data: PropTypes.string.isRequired
};
