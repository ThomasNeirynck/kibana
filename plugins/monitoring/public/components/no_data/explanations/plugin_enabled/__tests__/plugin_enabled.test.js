import React from 'react';
import { render } from 'enzyme';
import { ExplainPluginEnabled } from '../plugin_enabled';

describe('ExplainPluginEnabled', () => {
  test('should explain about xpack.monitoring.enabled setting', () => {
    const component = render(
      <ExplainPluginEnabled
        property="xpack.monitoring.enabled"
        data="false"
        context="cluster"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
