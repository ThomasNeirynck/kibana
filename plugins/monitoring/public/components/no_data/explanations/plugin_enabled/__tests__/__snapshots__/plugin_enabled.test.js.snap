// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplainPluginEnabled should explain about xpack.monitoring.enabled setting 1`] = `
<p>
  We checked the cluster settings and found that 
  <span
    class="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingLarge euiCodeBlock--inline"
  >
    <code
      class="euiCodeBlock__code"
    >
      xpack.monitoring.enabled
    </code>
  </span>
   is set to 
  <span
    class="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingLarge euiCodeBlock--inline"
  >
    <code
      class="euiCodeBlock__code"
    >
      false
    </code>
  </span>
   set, which disables monitoring. Removing the 
  <span
    class="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingLarge euiCodeBlock--inline"
  >
    <code
      class="euiCodeBlock__code"
    >
      xpack.monitoring.enabled: false
    </code>
  </span>
   setting from your configuration will put the default into effect and enable Monitoring.
</p>
`;
